### DISCUSSION TITLE
Is the web really the best choice of platform?

### DISCUSSION THREAD

**Discussion by wilsonowilson** [DevsHelpDevs/DevsHelpDevs#8] [General]
Deploying the project on the web may be a better option for a few reasons:

- Accessing a website may be easier for developers in poor countries.
- A website wouldn't take up much storage on a device.
- The site would work regardless of the OS, making it easy for devs to use any device they can to access it.

But then again, deploying the project as an app could also have a few benefits: 

- _Possibly_ faster development time because majority of the people in the immediate community are mobile developers.
- Most developers who struggle with tools should have a basic Android device, so installing an app may be convenient.
- More features available (Push Notifications, better connections to the system etc)

**Comment by sameeerkashyap** [comment]
Agree with your points, but the first three points seem more important, and although the development of an app maybe fast, deployment maybe worse,
-  complying with the app store, play store policies 
- waiting for their approval
-  making changes they suggest
- difficult integration of payment gateways 
- accounting for different country data policies, GDPR, etc.

This will make the process harder and frustrating. 
The web is a free place and the best one for more accessibility. 
and deploying is much easier especially with a wide array of options being free or very little price.

Maintenance and damage control is much easier as the website can be just taken down for a while.

For the features, PWAs can also send push notifications just like an app. 

And this might be a good learning opportunity for the community to gain some web dev experience, and if Flutter web becomes stable soon, it might be a better opportunity.

**Reply by biggaji** [reply]
I support @Sameerkash Points.
Deploying the project on the web would be the best option.
The web app would be faster to be develop than an app, and it can be easily maintained too.

**Comment by escamoteur** [comment]
I think we need a web app for the simple reason that this project relies on publicity and nothing is easier to share on social media than a simple link. To attract donors it has to have a great Desktop UI so you can easily access it from your workplace PS.

Mit might make sense to create an app for people applying for a donation but even there I think a PWA is more flexible.

**Reply by Jhanbury** [reply]
I agree an app in this scenario would be a very nice v2+ addition once the site has some usage.
A PWA makes a lot of sense as a placeholder if a mobile presence is really needed for the first version.

**Reply by escamoteur** [reply]
I actually currently tent a bit more for an app for applying as it makes adding pictures really easy and also let us store some unique ID on the device to detect of someone tries to apply multiple times. 
I would also wait with this decision till we know a bit better what experiences people who want to contribute have

**Reply by StephenLadoye** [reply]
For publicity, to attract donors and payment integration web app is the best option for a start

**Reply by 805702** [reply]
> I actually currently tent a bit more for an app for applying as it makes adding pictures really easy and also let us store some unique ID on the device to detect of someone tries to apply multiple times.
> I would also wait with this decision till we know a bit better what experiences people who want to contribute have

Very good idea... but i think the issue with respect to people applying many times is not an issue anymore. so we can get over that and take a web deployment. I thing it's the best route to go.
